skglm/estimators.py

```diff
@@ -492,8 +500,8 @@ class WeightedLasso(LinearModel, RegressorMixin):
     """
 
     def __init__(self, alpha=1., weights=None, max_iter=50, max_epochs=50_000, p0=10,
```

Comment:
**Suggestion:**

It appears that ``Lasso``, and `WeightedLasso` share a lot of similarities, noticeably in handling the case of weights=None for ``WeighedLasso``.

We shall think of merging both of them into one estimator.

WDYT @PABannier @mathurinm ?

Reply:
Indeed! actually we could even have `ElasticNet` be the base class for `Lasso` (l1_ratio = 1), and `Lasso` the base class for `WeightedLasso`, as in `scikit-learn`. Anyway, if you find it relevant, open a new issue and let's discuss it there ;)